AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -576,24 +578,24 @@ module {:options "/functionSyntax:4" } InternalInitializeMutation {
         CreateTime := timestamp,
         CiphertextBlob := [0] // [0] is a temporary place holder, but we should fix this by creating an internal type
       );
-      var SignedMutationIndex :- SystemKeyHandler.SignIndex(newIndex, systemKey);
+      var signedMutationIndex :- SystemKeyHandler.SignIndex(newIndex, systemKey);
+      assert CommitmentAndIndex.ValidIndex?(signedMutationIndex);
       // -= Write Mutation Index, conditioned on Mutation Commitment
       var throwAway2? := storage.WriteMutationIndex(
         KeyStoreTypes.WriteMutationIndexInput(
           MutationCommitment := commitment,
-          MutationIndex := SignedMutationIndex
+          MutationIndex := signedMutationIndex
         ));
       // TODO-Mutations-FF :: Ideally, we would diagnosis the Storage Failure.
       // What Condition Check failed?
       var _ :- throwAway2?.MapFailure(e => Types.Error.AwsCryptographyKeyStore(e));
       mutatedBranchKeyItems := mutatedBranchKeyItems
       + [Types.MutatedBranchKeyItem(ItemType := "Mutation Index: " + commitment.UUID, Description := "Created")];
     } else {
-      var commitmentAndIndex :- Mutations.ValidateCommitmentAndIndexStructures(
-        Token,
-        commitment,
-        index.value);
-      var indexIsVerified :- SystemKeyHandler.VerifyIndex(commitmentAndIndex.Index, systemKey);
+      var commitmentAndIndex := CommitmentAndIndex.CommitmentAndIndex(commitment, index.value);
+      :- CommitmentAndIndex.MutationItemsValidIDs(commitmentAndIndex);
+      :- CommitmentAndIndex.MutationItemsValidUTF8(commitmentAndIndex);
+      var indexIsVerified :- SystemKeyHandler.VerifyIndex(commitmentAndIndex.index, systemKey);
```

Comment:
Should we validate `token.Identifier == index.Identifier`?

Reply:
Actually, what we really need is a pre-condition on the Commitment and InitializeMutation input having the same branch-key-id.
After that, `CommitmentAndIndex.MutationItemsValidIDs` will take care of the rest.